transaction/core/src/blockchain/block_contents.rs

```diff
@@ -33,57 +31,35 @@ impl BlockContents {
         }
     }
 
-    /// The Blake2B256 digest of `self`.
+    /// The Merlin digest of `self`.
```

Comment:
Just thinking out loud: At one point, we moved from SHA3 to Blake2B because Blake2B was faster. Switching to Merlin (specifically, STROBE) roughly brings us back to SHA3. I don't think performance will be an issue here, and it feels like the correctness hygiene we get from Merlin transcripts is much more important.

Reply:
yeah -- STROBE is lower-level than SHA3, so it might be somewhat faster than SHA3. I don't really know -- we can do some profiling